crates/edr_provider/src/error.rs

```diff
@@ -314,31 +314,31 @@ pub struct TransactionFailure {
     pub data: String,
     #[serde(skip)]
     pub solidity_trace: Trace,
-    pub transaction_hash: B256,
+    pub transaction_hash: Option<B256>,
 }
 
 impl TransactionFailure {
     pub fn from_execution_result(
         execution_result: &ExecutionResult,
-        transaction_hash: &B256,
+        transaction_hash: Option<&B256>,
         solidity_trace: &Trace,
     ) -> Option<Self> {
         match execution_result {
             ExecutionResult::Success { .. } => None,
             ExecutionResult::Revert { output, .. } => Some(Self::revert(
                 output.clone(),
-                *transaction_hash,
+                transaction_hash.cloned(),
```

Comment:
Small nit: `cloned` could be changed to `copied` here and on line 335 to make it more expressive, but it's not a blocker to merge for me, since the clone will copy anyway under the hood.